### PR TITLE
fixed restrat btn hidden on game-over

### PR DESCRIPTION
**Pull request by ETLDevs**

### CODE REVIEW

File: game.js [hunk 1]
```diff
@@ -36,6 +22,75 @@ const saveStates = (playerState, savedState) => {
   });
 };
 
+const createBoard = (size) => {
+const gameContainer =  document.querySelector('.game');
```
Comment (shlomogcap) [inline game.js:26]: use prettier in order to make the code more clear with indentations etc.

File: game.js [hunk 2]
```diff
@@ -36,6 +22,75 @@ const saveStates = (playerState, savedState) => {
...
+gameContainer.classList.remove("hidden");
+if(game.boardSize !== "")
```
Comment (shlomogcap) [inline game.js:28]: always use {} inside if statements

File: game.js [hunk 3]
```diff
@@ -36,6 +22,75 @@ const saveStates = (playerState, savedState) => {
...
+gameContainer.classList.remove(game.boardSize);
```
Comment (shlomogcap) [inline game.js:29]: what is the purpose of this line? there is class with size? it doesn't make sense

File: game.js [hunk 4]
```diff
@@ -36,6 +22,75 @@ const saveStates = (playerState, savedState) => {
...
+gameContainer.innerHTML = "";
+for(let i = 0; i < (size**2); i++){
+  const cell = document.createElement('div');
+  cell.classList.add(`grid-cell`);
+  cell.dataset.value = i;
+  gameContainer.appendChild(cell);
+  game.boardSize =`x${size}`
```
Comment (shlomogcap) [inline game.js:36]: why this statement is inside the loop?

File: game.js [hunk 5]
```diff
@@ -36,6 +22,75 @@ const saveStates = (playerState, savedState) => {
...
+}
+gameContainer.classList.add(game.boardSize);
```
Comment (shlomogcap) [inline game.js:38]: why not use `size` param itself?

File: game.js [hunk 6]
```diff
@@ -36,6 +22,75 @@ const saveStates = (playerState, savedState) => {
...
+};
+
+const transposeMatrix = (matrix) => {
```
Comment (shlomogcap) [inline game.js:41]: try to move all this logic into another module (another file. e.g. getSolutionsState.js) and import if from there (i.e. 'import {getSolutions} from "./getSolutionsState.js" ')... it'll clean your file and make it more readable

File: game.js [hunk 7]
```diff
@@ -36,6 +22,75 @@ const saveStates = (playerState, savedState) => {
...
+};
+
+document.querySelectorAll(".board-btn").forEach((btn) => {
+  btn.addEventListener("click", () => {
+const boardSize = btn.dataset.board;
+ createBoard(boardSize);
+ getSolutions(boardSize);
+ 
+ document.querySelectorAll("button").forEach((btn) => {
+  (btn.classList.contains("hidden")) ?
```
Comment (shlomogcap) [inline game.js:87]: why not use toggle?

File: game.js [hunk 8]
```diff
@@ -127,15 +185,15 @@ document.querySelector(".save-game").addEventListener("click", function () {
...
+    if (savedStates[0].length > 0) {
     game.xState = [];
     game.oState =[];
     game.xState.push(...savedStates[0]);
     game.oState.push(...savedStates[1]);
     document.querySelectorAll(".grid-cell").forEach((cell) => {
       cell.classList.remove("disabled", "x", "o");
-      if (game.xState.includes(cell.dataset.value)) cell.classList.add("x");
-      if (game.oState.includes(cell.dataset.value)) cell.classList.add("o");
+      if (game.xState.includes(cell.dataset.value)) cell.classList.add("x", "disabled");
```
Comment (shlomogcap) [inline game.js:195]: use {}

File: game.js [hunk 9]
```diff
@@ -127,15 +185,15 @@ document.querySelector(".save-game").addEventListener("click", function () {
...
+      if (game.oState.includes(cell.dataset.value)) cell.classList.add("o", "disabled");
```
Comment (shlomogcap) [inline game.js:196]: use {}

File: game.js [hunk 10]
```diff
@@ -149,12 +207,22 @@ document.querySelectorAll(".restart").forEach((btn) => {
       document.querySelector(".game-over").classList.remove("visible");
       game.gamesCounter++;
     }
+    document.querySelectorAll("button").forEach((btn) => {
+      (btn.classList.contains('hidden')) ?
```
Comment (shlomogcap) [inline game.js:211]: use toggle

File: styles.css [hunk 11]
```diff
@@ -1,12 +1,14 @@
-#game {
+.game {
     position: absolute;
     top: 50%;
     left: 50%;
     transform: translate(-50%, -50%);
     display: grid;
     grid-template-columns: repeat(3, auto);
```
Comment (shlomogcap) [inline styles.css:7]: ### use cssVariables:

- [ ] add ":root" section with --board-size variable:
```
:root{
   --board-size: 3;
}
```
- [ ] update you style to something like:
`grid-template-columns: repeat(var(--board-size), auto);`
- [ ] then in you js you can do something like:
```
document.documentElement.style.setProperty('--board-size', YOUR_NEW_SIZE);
```

---
see this [post for example](https://css-tricks.com/updating-a-css-variable-with-javascript/)

File: game.js (hunk 3 above)
Comment (ETLDevs) [reply game.js:29]: I just taking the class name of 'game-container' from boardSize(x3/x5)

File: game.js (hunk 4 above)
Comment (ETLDevs) [reply game.js:36]: my mistake!

File: game.js (hunk 5 above)
Comment (ETLDevs) [reply game.js:38]: beacause I use it in another function when I want to remove the class

File: game.js (hunk 6 above)
Comment (ETLDevs) [reply game.js:41]: ok! ill try

File: game.js (hunk 5 above)
Comment (shlomogcap) [reply game.js:38]: jot you. so just save it to variable (e.g. const boardSizeClass = `x${size}`) , and the use it when you need it inside this scope

File: game.js (hunk 2 above)
Comment (ETLDevs) [reply game.js:28]: even if its for just one line?
writing if(...) return .... is better?

Comment (shlomogcap) [reply game.js:28]: always use curly braces for consistency

File: game.js (hunk 7 above)
Comment (ETLDevs) [reply game.js:87]: youre rigtht ill use it!

File: game.js (hunk 2 above)
Comment (ETLDevs) [reply game.js:28]: ok!

File: game.js (hunk 10 above)
Comment (ETLDevs) [reply game.js:211]: done!

File: game.js (hunk 5 above)
Comment (ETLDevs) [reply game.js:38]: you mean to make it global?

Comment (shlomogcap) [reply game.js:38]: no. only at the very beginning of `createBoard` function

File: game.js [hunk 12]
```diff
@@ -23,72 +24,31 @@ const saveStates = (playerState, savedState) => {
...
- 
- document.querySelectorAll("button").forEach((btn) => {
-  (btn.classList.contains("hidden")) ?
-  btn.classList.remove("hidden") :
-  btn.classList.add("hidden");
-});
-});
+      const boardSize = btn.dataset.board;
+      createBoard(boardSize);
+      getSolutions(boardSize);
```
Comment (shlomogcap) [inline game.js:47]: `game.boardSize = getSolutions(boardSize);`

File: game.js [hunk 13]
```diff
@@ -9,6 +9,7 @@ export const game = {
   gamesCounter: 1,
   winnings: {},
   winningStates: [],
+  alerts: [["The board still empty!"], ["No winnings yet"], ["No saved game"]]
```
Comment (shlomogcap) [inline game.js:12]: i like the idea of saving alerts in constant global scope... 
but why this nested array? you can simply do : ['msg1','msg2'] no need this [['msg1'],['msg2']] approach.
better idea will be to put the in object:
alert:{
emptyBoard:'the board still empty',
noWinningYet:'No winning yet'... etc
}
this way you don't need to hardcoded the index of the array ...

**Review by shlomogcap — APPROVED** [review]
nice job... lets going forward with this... i'll approve this PR ... regarding the latest comments , i'll move them to new issues so we can take them later...

File: game.js (hunk 13 above)
Comment (ETLDevs) [reply game.js:12]: right! i'll go with the alerts object

File: styles.css (hunk 11 above)
Comment (ETLDevs) [reply styles.css:7]: done! works like a charm..

**Pull request merged**